lib/setupHooks/inheritCargoArtifactsHook.sh

```diff
@@ -66,27 +65,15 @@ inheritCargoArtifacts() {
         --times \
         --chmod=u+w \
         --executability \
-        --exclude 'deps/*.rlib' \
-        --exclude 'deps/*.rmeta' \
+        --exclude '*/deps/*' \
         --exclude '.cargo-lock' \
         "${preparedArtifacts}/" \
         "${cargoTargetDir}/"
-
-      local linkCandidates=$(mktemp linkCandidatesXXXX.txt)
-      find "${preparedArtifacts}" \
-        '(' -path '*/deps/*.rlib' -or -path '*/deps/*.rmeta' ')' \
-        -printf "%P\n" \
-        >"${linkCandidates}"
-
-      # Next create any missing directories up front so we can avoid redundant checks later
-      cat "${linkCandidates}" \
-        | xargs --no-run-if-empty -n1 dirname \
-        | sort -u \
-        | (cd "${cargoTargetDir}"; xargs --no-run-if-empty mkdir -p)
-
-      # Lastly do the actual symlinking
-      cat "${linkCandidates}" \
-        | xargs -P ${NIX_BUILD_CORES} -I '##{}##' ln -s "${preparedArtifacts}/##{}##" "${cargoTargetDir}/##{}##"
+      
+      symlinks="${cargoArtifacts}/symlinks.tar"
+      if [[ -f "${symlinks}" ]]; then
+        tar -xf "${symlinks}" -C "${cargoTargetDir}" --strip-components=1
```

Comment:
Would still compress.

Reply:
Question: Now that everything is a symlink, but the symlink is invisible to Nix because it's in a tar archive, what is going to prevent Nix from GCing the step being linked to?

Reply:
@dpc's observation here is correct. Nix will perform a naive scan of the results for any store paths, but will not notice them if they are behind a tarball, meaning a store GC would end up invalidating references.

Reply:
An uncompressed tar file is just the uncompressed data with some special headers - no fancy behaviour. Likewise, the content of a symlink is just the path. My understanding is that Nix will have no trouble at all understanding the dependency. Hex editor image placed below.

![image](https://github.com/ipetkov/crane/assets/42533/0f49dda2-b4c3-47d1-9182-5b05d141ae2f)

Reply:
However, when compressed, I'd expect it's hard for nix to find what it's looking for, hence the explicit choice to not compress.

Reply:
That might work, but I wonder: would compressing + putting 1 symlink somewhere in the output (outside of what is copied to `./target`), keep the GC happy? Aren't all these paths pointing to the same things in nix store (or the thing it points to)?

Reply:
it'd definitely work - mostly was going for simplicity vs optimality. A 345MB deps dir on my machine ends up with a 900kB symlinks.tar. Might be more clean to explicitly add a symlink though from the perspective of making the purpose clear.

Reply:
As long as there is a single line somewhere in the output (which can be a separate text file and not necessarily the tarball) which includes `/nix/store/...-blah` then Nix will consider that path a dependency of the derivation. So it's okay (as far as the GC is concerned) to create a tarball of symlinks, so long as there is some other symlink or text entry which points to the same "parent" directory that the tarball's symlinks will end up pointing to